### PR TITLE
ICS Input & Output for a Schedule 

### PR DESCRIPTION
**Pull request by kenneth-miura**
Mostly done. I have one function I can't find a good name for, and I'm having trouble figuring out how to test the function. Also need to have info regarding a Session's term code (F,S,Y) & it's course code & lec/tutorial code (To make a title).

### CODE REVIEW

**Comment by kenneth-miura** [comment]
Ok so it's done now but I have no idea how to test it, since it exports to a file.

Originally I wanted to use the fact the Library for ICS lets you pass it a `Writer` to give `outputScheduleICS` a `StringWriter` and check correctness against a String, but that doesn't work b/c there's no way to get the `String` from `StringWriter` without returning it. Since returning a String from `outputScheduleICS` doesn't make sense in the normal case, where you use `FileWriter` to write to an ICS file, I'm stuck.

File: src/test/java/ScheduleExporterTest.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        Scheduler scheduler = new Scheduler();
+        inPersonSchedule = scheduler.createBasicSchedule(inPerson);
+        onlineSchedule = scheduler.createBasicSchedule(online);
+
+        exporter = new ScheduleExporter();
+    }
+
+    @Test(timeout = 100)
+    public void testSessionExportInPerson() {
+        // TODO: Fill this in later
```
Comment (EclipseIsDead) [inline src/test/java/ScheduleExporterTest.java:30]: Check [this](https://stackoverflow.com/questions/3840356/how-to-test-in-java-that-a-class-implements-serializable-correctly-not-just-is) link out. Essentially you go:
`new ObjectOutputStream(new ByteArrayOutputStream()).writeObject(myObject);`
and make sure no error is thrown.

File: src/main/java/workers/ScheduleExporter.java [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+                termEndDate = WINTER_SEMESTER_END_DATE;
+                break;
+            case "Y":
+                termStartDate = FALL_SEMESTER_START_DATE;
+                termEndDate = WINTER_SEMESTER_END_DATE;
+                break;
+            default:
+                break;
+        }
+        Date finalDay =
```
Comment (EclipseIsDead) [inline src/main/java/workers/ScheduleExporter.java:144]: is this spotlessApply formatted?

File: src/main/java/workers/ScheduleExporter.java [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+            case "S":
+                termStartDate = WINTER_SEMESTER_START_DATE;
+                termEndDate = WINTER_SEMESTER_END_DATE;
+                break;
+            case "Y":
+                termStartDate = FALL_SEMESTER_START_DATE;
+                termEndDate = WINTER_SEMESTER_END_DATE;
+                break;
+            default:
+                break;
```
Comment (EclipseIsDead) [inline src/main/java/workers/ScheduleExporter.java:142]: i don't think default -> break is good practice

File: src/main/java/workers/ScheduleExporter.java [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+import net.fortuna.ical4j.model.DateTime;
+import net.fortuna.ical4j.model.Recur;
+import net.fortuna.ical4j.model.component.VEvent;
+import net.fortuna.ical4j.model.property.*;
+import net.fortuna.ical4j.util.RandomUidGenerator;
+import net.fortuna.ical4j.util.UidGenerator;
+import net.fortuna.ical4j.validate.ValidationException;
+
+/** Use Case class for exporting a Schedule into an ICS file */
+public class ScheduleExporter {
```
Comment (EclipseIsDead) [inline src/main/java/workers/ScheduleExporter.java:27]: make sure ur using static

File: src/main/java/workers/ScheduleExporter.java [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+            outputDirectory.mkdir();
+        }
+    }
+
+    /**
+     * @param year current year
+     * @param month current month
+     * @return The year in which the current Academic Year started
+     */
+    private int getStartYear(int year, int month) {
```
Comment (EclipseIsDead) [inline src/main/java/workers/ScheduleExporter.java:51]: condense this into a ternary

File: src/main/java/workers/ScheduleExporter.java [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+     * {@code writer} defaults to {@link FileWriter}
+     *
+     * @see ScheduleExporter#outputScheduleICS(Schedule, Writer)
+     */
+    public void outputScheduleICS(Schedule schedule) {
+        try {
+            Writer writer = new FileWriter(outputDirectory.getAbsolutePath().concat("/temp.ics"));
+            outputScheduleICS(schedule, writer);
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (EclipseIsDead) [inline src/main/java/workers/ScheduleExporter.java:69]: you could try also printing the schedule that did this error if u want

**Review by EclipseIsDead — CHANGES_REQUESTED** [review]
Overall solid work, gave a quick look

File: src/main/java/workers/ScheduleExporter.java (hunk 2 above)
Comment (kenneth-miura) [reply src/main/java/workers/ScheduleExporter.java:144]: yeah idk why it looks like that

File: src/main/java/workers/ScheduleExporter.java (hunk 4 above)
Comment (kenneth-miura) [reply src/main/java/workers/ScheduleExporter.java:27]: y

**Review by EclipseIsDead — APPROVED** [review]
LGTM!

**Review by lorena-b — APPROVED** [review]

**Pull request merged**